### PR TITLE
Move simulators to common submodule

### PR DESCRIPTION
**Pull request by fabiocruz**
This PR moves all simulator classes to a common submodule `simulators`. Individual simulators are now accessible through, e.g.:

```python
import inductiva
simulator = inductiva.simulators.OpenFOAM()
```

I renamed a previous submodule called `simulation` that contained the base `Simulator` class to `simulators` and added all simulators to it. 

For further context, see the discussion on https://github.com/inductiva/inductiva/pull/523.

This PR touches on a large number of files, but the changes are small and mostly related to updating import statements. Note that I haven't fixed imports to follow the Google Python style guide yet, as I think a thorough revision of those should be made on a per-scenario/per-submodule basis.

### CODE REVIEW

**Review by IvanPombo — APPROVED** [review]
Massive @fabiocruz! Thank you so much! We really need a pipeline to test each scenario and simulator for these picky changes over a bunch of files.

File: demos/py_scripts/gromacs.py
Comment (IvanPombo) [inline demos/py_scripts/gromacs.py:1]: Any reason to remove this?

Comment (fabiocruz) [reply demos/py_scripts/gromacs.py:1]: Yep, this script uses a deprecated interface of the `run` method.

**Review by luispcunha — APPROVED** [review]
lgtm, thanks @fabiocruz

**Pull request merged**